examples/software-device/rs-software-device.cpp

```diff
@@ -0,0 +1,185 @@
+                pixels_color.data()[i*frame.x * 4 + j + 2] = 185;
+            }
+        }
+    frame.frame = pixels_color;
+
+    return frame;
+}
+
+void fill_synthetic_depth_data(void* data, int w, int h, int bpp, float wave_base)
+{
+    for (int i = 0; i < h; i++)
+    {
+        for (int j = 0; j < w; j++)
+        {
+            auto d = 2 + 0.2 * 0.5 * (1 + sin(wave_base + j / 50.f));
+            ((uint16_t*)data)[i*w + j] = (int)(d * 0xff);
+        }
+    }
+}
+
+int main(int argc, char * argv[]) try
+{
+    const int W = 640;
+    const int H = 480;
+    const int BPP_D = 2;
+
+    rs2_intrinsics depth_intrinsics{ W, H, W / 2, H / 2, W , H , RS2_DISTORTION_BROWN_CONRADY ,{ 0,0,0,0,0 } };
+
+    auto texture = create_synthetic_texture();
+
+    rs2_intrinsics color_intrinsics = { texture.x, texture.y, 
+        (float)texture.x / 2, (float)texture.y / 2,
+        (float)texture.x / 2, (float)texture.y / 2,
+        RS2_DISTORTION_BROWN_CONRADY ,{ 0,0,0,0,0 } };
+
+   
+    //==================================================//
+    //           Declare Software-Only Device           //
+    //==================================================//
+    
+    software_device dev; // Create software-only device
+
+    auto depth_s = dev.add_sensor("Depth"); // Define single sensor 
+    auto color_s = dev.add_sensor("Color"); // Define single sensor 
+
+
+    depth_s.add_video_stream({  RS2_STREAM_DEPTH, 0, 0, 
+                                W, H, 60, BPP_D, 
+                                RS2_FORMAT_Z16, depth_intrinsics });
+
+    depth_s.add_read_only_option(RS2_OPTION_DEPTH_UNITS, 0.001f);
+
+
+    color_s.add_video_stream({  RS2_STREAM_COLOR, 0, 1, texture.x, 
+                                texture.y, 60, texture.bpp, 
+                                RS2_FORMAT_RGBA8, color_intrinsics });
+
+    dev.create_matcher(DLR_C);
+    syncer sync;
+
+    depth_s.start(sync);
+    color_s.start(sync);
+
+    auto depth_stream = depth_s.get_stream_profiles().front();
+    auto color_stream = color_s.get_stream_profiles().front();
+
+    depth_stream.register_extrinsics_to(color_stream, { { 1,0,0,0,1,0,0,0,1 },{ 0,0,0 } });
+
+
+    window app(1280, 1500, "RealSense Capture Example");
+    state app_state;
+
+    register_glfw_callbacks(app, app_state);
+
+    rs2::colorizer color_map; // Save colorized depth for preview
+
+    auto last = std::chrono::high_resolution_clock::now();
+
+    float wave_base = 0.f;
+    pointcloud pc;
+    points p;
+    int ind = 0;
+
+    std::vector<uint8_t> pixels_depth(W * H * BPP_D, 0);
+
+    while (app) // Application still alive?
+    {
+        draw_text(50, 50, "This point-cloud is generated from a synthetic device:");
+        fill_synthetic_depth_data((void*)pixels_depth.data(), W , H , BPP_D, wave_base);
+
+        auto now = std::chrono::high_resolution_clock::now();
+        if (std::chrono::duration_cast<std::chrono::milliseconds>(now - last).count() > 1)
```

Comment:
`if( (now - last) > std::chrono::milliseconds(1))` is preferred.
Even better is:
```cpp
const auto some_threashold_with_proper_name = std::chrono::milliseconds(1);
if( (now - last) > some_threashold_with_proper_name )` 
```